pkg/k8scontext/context.go

```diff
@@ -171,6 +171,16 @@ func (c *Context) Run(stopChannel chan struct{}, omitCRDs bool, envVariables env
 	return nil
 }
 
+// GetAGICPod returns the pod with specified name and namespace
+func (c *Context) GetAGICPod(envVariables environment.EnvVariables) (*v1.Pod, error) {
+	pod, err := c.kubeClient.CoreV1().Pods(envVariables.AGICPodNamespace).Get(envVariables.AGICPodName, metav1.GetOptions{})
```

Comment:
Why don't we move this one line somewhere above this: https://github.com/Azure/application-gateway-kubernetes-ingress/blob/master/cmd/appgw-ingress/main.go#L128
And there will be no need then to create this new exported method on the K8s Context struct.